modules/speech-to-text-async-impl/src/main/java/org/opencastproject/speechtotext/async/impl/SpeechToTextWorkflowSchedulerQuartz.java

```diff
@@ -0,0 +1,403 @@
+  public static final String JOB_NAME = "org-opencast-stt-wf-scheduler-job";
+  public static final String SCANNER_NAME = "STT Worflow Scheduler Quartz";
+  public static final String TRIGGER_GROUP = "org-opencast-stt-wf-scheduler-trigger-group";
+  public static final String TRIGGER_NAME = "org-opencast-stt-wf-scheduler-trigger";
+
+  // === Configuration options
+  /* Number of seconds to abandon retrying attaching subtitles after they are generated. */
+  public static final String ABANDON_AFTER_SECS = "abandon-after-secs";
+  /* Workflow definition to use to attach the subtitles */
+  public static final String WORKFLOW = "workflow";
+
+  // === Default values
+  /* Default number of seconds to abandon retrying attaching subtitles. */
+  private static final long DEFAULT_ABANDON_AFTER_SECS = 24 * 60 * 60;
+
+  /* Workflow definition to start to attach the subtitles */
+  private String attachWorkflowDef;
+  /* If STT jobs not finished after this interval, they won't be tracked anymore. */
+  private long abandonAfterMs = DEFAULT_ABANDON_AFTER_SECS * 1000;
+
+  private AssetManager assetManager;
+  private WorkflowService workflowService;
+  private SpeechToTextDatabase database;
+
+  // Only used by unit tests!
+  private Workflows wfUtil;
+
+  @Override
+  @Activate
+  protected void activate(ComponentContext cc) {
+    logger.info("Activating!");
+    super.activate(cc);
+  }
+
+  public SpeechToTextWorkflowSchedulerQuartz() {
+    try {
+      quartz = new StdSchedulerFactory().getScheduler();
+      quartz.start();
+      // create and set the job. To actually run it call schedule(..)
+      final JobDetail job = new JobDetail(getJobName(), getJobGroup(), Runner.class);
+      job.setDurability(false);
+      job.setVolatility(true);
+      job.getJobDataMap().put(JOB_PARAM_PARENT, this);
+      quartz.addJob(job, true);
+    } catch (org.quartz.SchedulerException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void updated(Dictionary<String, ?> properties) throws ConfigurationException {
+    logger.info("Updating!");
+    unschedule();
+
+    if (properties != null) {
+      logger.debug("Updating configuration...");
+
+      Option<Boolean> optEnabled = OsgiUtil.getOptCfgAsBoolean(properties, PARAM_KEY_ENABLED);
+      setEnabled(optEnabled.isSome() && optEnabled.get());
+      if (!isEnabled()) {
+        logger.info("Speech to text workflow scheduler quartz job is disabled");
+        return;
+      }
+
+      String cronExpression = (String) properties.get(PARAM_KEY_CRON_EXPR);
+      if (StringUtils.isBlank(cronExpression) || !CronExpression.isValidExpression(cronExpression)) {
+        throw new ConfigurationException(PARAM_KEY_CRON_EXPR, "Cron expression must be valid");
+      }
+      setCronExpression(cronExpression);
+      logger.info("{} : {}", PARAM_KEY_CRON_EXPR, cronExpression);
+
+      attachWorkflowDef = (String) properties.get(WORKFLOW);
```

Comment:
So if this key is unset we just roll with it?  What's the usecase for *not* having a workflow ID set?